### PR TITLE
[MapView][Part 1] Move GestureOptions into GestureManager

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>GestureOptions are owned by GestureManager directly</changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR makes the `GestureManager` directly hold and expose the `GestureOptions`. It allows a developer to directly manipulate the options instead of going via `update`.

### CODE REVIEW

**Review by jmkiley — APPROVED** [review]

File: Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+internal protocol CameraManagerProtocol: AnyObject {
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift:3]: Nit: Can we group `CameraManagerProtocol` in the file path for Camera, not Gestures

File: Sources/MapboxMaps/Gestures/GestureHandlers/GestureHandler.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+internal class GestureHandler {
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Gestures/GestureHandlers/GestureHandler.swift:3]: nit: Move internal modifiers to each item below as opposed to on the class

File: Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift (hunk 1 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift:3]: Lets do this afterwards.

File: Sources/MapboxMaps/Gestures/GestureHandlers/GestureHandler.swift (hunk 2 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Gestures/GestureHandlers/GestureHandler.swift:3]: This was a simple move, so this existed earlier.

File: Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift (hunk 1 above)
Comment (neelmistry94) [reply Sources/MapboxMaps/Gestures/CameraManagerProtocol.swift:3]: But its the protocol on the camera, like it belongs to the camera from what I see.

**Pull request merged**